zapcore/sampler_test.go

```diff
@@ -40,103 +40,97 @@ func fakeSampler(lvl LevelEnabler, tick time.Duration, first, thereafter int) (F
 	return fac, &logs
 }
 
-func buildExpectation(level Level, nums ...int) []observer.LoggedEntry {
-	var expected []observer.LoggedEntry
-	for _, n := range nums {
-		expected = append(expected, observer.LoggedEntry{
-			Entry:   Entry{Level: level},
-			Context: []Field{makeInt64Field("iter", n)},
-		})
+func assertSequence(t testing.TB, logs []observer.LoggedEntry, lvl Level, seq ...int64) {
+	seen := make([]int64, len(logs))
+	for i, entry := range logs {
+		require.Equal(t, "", entry.Message, "Message wasn't created by writeSequence.")
+		require.Equal(t, 1, len(entry.Context), "Unexpected number of fields.")
+		require.Equal(t, lvl, entry.Level, "Unexpected level.")
+		f := entry.Context[0]
+		require.Equal(t, "iter", f.Key, "Unexpected field key.")
+		require.Equal(t, Int64Type, f.Type, "Unexpected field type")
+		seen[i] = f.Integer
 	}
-	return expected
+	assert.Equal(t, seq, seen, "Unexpected sequence logged at level %v.", lvl)
```

Comment:
This approach makes the failure output from testify much more useful - instead of seeing a huge diff of Entry structs, we just see the diff in sequence numbers.